torch/csrc/distributed/c10d/quantization/quantization.cpp

```diff
@@ -0,0 +1,93 @@
+#include <torch/csrc/distributed/c10d/quantization/quantization.h>
```

Comment:
These seem like general quantization libraries, should these be in c10d or some general torch folder which has rest of quant stuff?

Reply:
That's good point. I initially had them in the c10d but I didn't see any other quantization code in that directory that's why I moved them all to a separate folder.

Reply:
I feel like the reason we might leave this in distributed folder is, there's a limitation different backends (MPI, Gloo) for not accepting bfp16 dtyped tensor, in torch quantization area user can do things like `to(torch.bfloat16)` and `to(torch.float32)`  to do quantization/dequant. Once we figure out the way to support bfp16 tensor in the comm backend, we can use `to` instead of custom kernels.